src/validation/obfl.rng

```diff
@@ -826,6 +829,21 @@
       <data type="IDREF"/>
     </attribute>
   </define>
+  <define name="toc-entry-on-resumed">
+    <element name="toc-entry-on-resumed">
+      <ref name="attlist-toc-entry-on-resumed"/>
+      <zeroOrMore>
+        <ref name="blockContents"/>
+      </zeroOrMore>
+    </element>
+  </define>
+  <define name="attlist-toc-entry-on-resumed" combine="interleave">
+    <attribute name="id-range">
+        <data type="string">
+            <param name="pattern">\[(\w|_)+,(\w|_)*((\w|_)\]|\))</param>
```

Comment:
An equivalent but better way to write this regex is: "\[\w+,(\w+\]|\w*\))".

It is a bit too restrictive though: not all valid IDREF match "\w+". An IDREF is a "Name":

    NameStartChar ::=   ":" | [A-Z] | "_" | [a-z] | \
                        [#xC0-#xD6] | [#xD8-#xF6] | [#xF8-#x2FF] | \
                        [#x370-#x37D] | [#x37F-#x1FFF] | [#x200C-#x200D] | \
                        [#x2070-#x218F] | [#x2C00-#x2FEF] | [#x3001-#xD7FF] | \
                         [#xF900-#xFDCF] | [#xFDF0-#xFFFD] | [#x10000-#xEFFFF]
    NameChar      ::=   NameStartChar | "-" | "." | [0-9] | \
                        #xB7 | [#x0300-#x036F] | [#x203F-#x2040]
    Name          ::=   NameStartChar (NameChar)*

Reply:
I have tested with the jing validator and it appeared that jing didn't consider an underscore to be part of `\w`. That's why I wrote everywhere `(\w|_)` instead of simply `\w`.

Reply:
Hmm weird. Than jing is wrong. But anyway, the other part of the comment is more important.

Reply:
Why not change the \w to [^,] (not comma).
The regexp then becomes: \[[^,]+,([^,]+\]|[^,]*\))

Reply:
If you mean no comma AND no brackets: yes that would be better. Still not accurate, but at least it will not reject valid IDREFs.